source/opt/instruction.cpp

```diff
@@ -735,5 +763,28 @@ bool Instruction::IsOpcodeSafeToDelete() const {
   }
 }
 
+void DebugScope::ToBinary(uint32_t type_id, uint32_t result_id,
+                          uint32_t ext_set,
+                          std::vector<uint32_t>* binary) const {
+  uint32_t num_words = 7;
+  uint32_t dbg_opcode = 23;
+  if (!lexical_scope_) {
```

Comment:
For this and the next check, it would be easier to read of you compare with the named constants.  Also use the accessor functions.

```
if (GetLexicalScope() == kNoDebugScope) {
```